File: validatedtextinputlayout/src/main/java/com/julianraj/validatedtextinputlayout/validator/RegexValidator.java
package com.julianraj.validatedtextinputlayout.validator;

import android.support.annotation.NonNull;

import com.julianraj.validatedtextinputlayout.ValidatedTextInputLayout;

import java.util.regex.Pattern;

/**
 * Validator to check if value satisfies given regular expression
 * @see BaseValidator
 */
public class RegexValidator extends BaseValidator {

    final String mRegex;

    /**
     * @param pRegex regular expression to check against
     * @param pErrorMessage error message to display if validation fails
     */
    public RegexValidator(@NonNull String pRegex, @NonNull String pErrorMessage) {
        super(pErrorMessage);
        mRegex = pRegex;
    }

    /**
     * @param pRegex regular expression to check against
     * @param pErrorMessage error message to display if validation fails
     * @param pCallback callback for validation event
     */
    public RegexValidator(@NonNull String pRegex, @NonNull String pErrorMessage,
                          ValidationCallback pCallback) {
        super(pErrorMessage, pCallback);
        mRegex = pRegex;
    }

    /**
     * Check if the associated {@link ValidatedTextInputLayout} has correct value.
     *
     * @param pText value associated with the input field
     * @return validity of the field
     */
    @Override
    public boolean isValid(String pText) {
        return Pattern.compile(mRegex, Pattern.CASE_INSENSITIVE).matcher(pText).find();
    }
}

File: validatedtextinputlayout/src/main/java/com/julianraj/validatedtextinputlayout/validator/RegexValidator.kt
package com.julianraj.validatedtextinputlayout.validator

import com.julianraj.validatedtextinputlayout.ValidatedTextInputLayout

import java.util.regex.Pattern

/**
 * Validator to check if value satisfies given regular expression
 *
 * @param regex regular expression to check against
 * @param errorMessage Error message that will be displayed if validation fails.
 * @param callback callback with validation result
 *
 * @see BaseValidator
 */
class RegexValidator(private val regex: String,
                     errorMessage: String,
                     callback: ((Boolean)-> Unit)? = null) : BaseValidator(errorMessage, callback) {

    /**
     * Check if the associated [ValidatedTextInputLayout] has correct value.
     *
     * @param text value associated with the input field
     * @return validity of the field
     */
    override fun isValid(text: String): Boolean {
        return Pattern.compile(regex, Pattern.CASE_INSENSITIVE).matcher(text).find()
    }
}
